app/modules/asset_groups/models.py

```diff
@@ -786,226 +816,298 @@ def build_detection_request(self, job_uuid, model):
             _external=True,
         )
 
-        ia_config_reader = IaConfig(current_app.config.get('CONFIG_MODEL'))
-        detector_config = ia_config_reader.get_named_detector_config(model)
-
-        assert 'start_detect' in detector_config
-
         model_config = {
-            'endpoint': detector_config['start_detect'],
             'jobid': str(job_uuid),
             'callback_url': f'houston+{callback_url}',
             'callback_detailed': True,
         }
+
+        ia_config_reader = IaConfig(current_app.config.get('CONFIG_MODEL'))
+        detector_config = ia_config_reader.get_named_detector_config(model)
         model_config.update(detector_config)
 
-        asset_urls = []
+        assets = []
         if 'updatedAssets' in self.sighting_config:
-            asset_urls = [
-                url_for('api.assets_asset_src_raw_by_id', asset_guid=guid, _external=True)
-                for guid in self.sighting_config['updatedAssets']
-            ]
+            for guid in self.sighting_config['updatedAssets']:
+                asset = Asset.query.get(guid)
+                if not asset:
+                    logging.warning(f'Asset ref {guid}, cannot find asset, skipping')
+                    continue
+                assets.append(asset)
         else:
             for filename in self.sighting_config.get('assetReferences'):
                 asset = self.asset_group.get_asset_for_file(filename)
                 if not asset:
                     logging.warning(f'Asset ref {filename}, cannot find asset, skipping')
                     continue
+                assets.append(asset)
 
-                asset_url = url_for(
-                    'api.assets_asset_src_raw_by_id',
-                    asset_guid=str(asset.guid),
-                    _external=True,
+        asset_sage_data = []
+
+        if preload:
+            # Ensure that the assets exist on Sage
+            for asset in assets:
+                asset.sync_with_sage(ensure=True)
+                asset_sage_data.append(
+                    (
+                        to_sage_uuid(asset.content_guid),
+                        asset.guid,
+                    )
                 )
-                if asset_url not in asset_urls:
-                    asset_urls.append(asset_url)
-
-        # Sort the Asset URLs so that when processing the response we know which
-        # content guid relates to which asset guid
-        asset_urls.sort()
-        model_config['image_uuid_list'] = json.dumps(
-            [f'houston+{asset_url}' for asset_url in asset_urls]
-        )
-        return model_config
+
+            asset_sage_data.sort(key=lambda data: data[0]['__UUID__'])
+        else:
```

Comment:
If `preload=False`, we send the Houston asset URLs to Sage, which will download the assets from Houston.  This is not preferred for two reasons: 1) We don't want to send the file bytes to Sage if it already has the asset (which Sage will have to download the file to determine if it already has the asset) and 2) The asset may have been deleted or access otherwise disabled between the time Sage received the request and actually attempts to download the files.  Whevener possible, we should front-load the sending of Assets and Annotations to Sage because Houston maintains control of the processing flow and can respond to errors in the foreground process